.github/workflows/checks.yaml

```diff
@@ -0,0 +1,184 @@
+name: Checks
+
+on:
+  push:
+    branches:
+      - main
+      - 2.*
+  pull_request: ~
+
+env:
+  # Also change CACHE_VERSION in the other workflows
+  CACHE_VERSION: 5
+  DEFAULT_PYTHON: 3.8
+  PRE_COMMIT_CACHE: ~/.cache/pre-commit
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  prepare-base:
```

Comment:
Shouldn't prepare-base be in its own file as it's used in checks as well as in tests ?

Reply:
It is isn't used in tests 😄 

I thought so as well, but it isn't. Which makes this separation so easy.